mindsdb/integrations/handlers/ray_serve_handler/ray_serve_handler.py

```diff
@@ -1,56 +1,69 @@
             raise Exception("Error: Please provide a URL for the training endpoint.")
         if not args['using'].get('predict_url'):
             raise Exception("Error: Please provide a URL for the prediction endpoint.")
 
     def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
-        # TODO: use join_learn_process to notify users when ray has finished the training process
-        args = args['using']  # ignore the rest of the problem definition
-        args['target'] = target
-        self.model_storage.json_set('args', args)
+        args = copy.copy(args['using'])
+        train_url = args.pop('train_url', None)
+        predict_url = args.pop('predict_url', None)
+
+        if not train_url or not predict_url:
+            raise Exception("Error: Both 'train_url' and 'predict_url' must be provided.")
+
+        self.model_storage.json_set('args', {
+            'train_url': train_url,
+            'predict_url': predict_url,
+            **args
+        })
+
+        payload = {
+            'df': df.to_json(orient='records'),
+            'target': target,
+            'args': args
+        }
+
         try:
-            resp = requests.post(args['train_url'],
-                                 json={'df': df.to_json(orient='records'), 'target': target},
-                                 headers={'content-type': 'application/json; format=pandas-records'})
-        except requests.exceptions.InvalidSchema:
-            raise Exception("Error: The URL provided for the training endpoint is invalid.")
+            resp = requests.post(
+                train_url,
+                json=payload,
+                headers={'content-type': 'application/json; format=pandas-records'}
+            )
+            resp.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            raise Exception(f"Error: Failed to communicate with the training endpoint. Details: {str(e)}")
 
-        resp = resp.json()
-        if resp['status'] != 'ok':
-            raise Exception("Error: Training failed: " + resp['status'])
+        response = resp.json()
+        if response.get('status') != 'ok':
+            raise Exception(f"Error: Training failed. Details: {response.get('status', 'Unknown error')}")
 
     def predict(self, df, args=None):
-        args = self.model_storage.json_get('args')  # override any incoming args for now
-        resp = requests.post(args['predict_url'],
-                             json={'df': df.to_json(orient='records')},
-                             headers={'content-type': 'application/json; format=pandas-records'})
+        args = self.model_storage.json_get('args')
+        resp = requests.post(
+            args['predict_url'],
+            json={'df': df.to_json(orient='records')},
+            headers={'content-type': 'application/json; format=pandas-records'}
+        )
         response = resp.json()
 
         target = args['target']
```

Comment:
I've tested with ray server and it has error in this line because target was removed in `create`

I've changed this code to it to make it work
```python
    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:

        args = args['using']  # ignore the rest of the problem definition
        args['target'] = target
        self.model_storage.json_set('args', args)

        train_url = args.pop('train_url', None)
        predict_url = args.pop('predict_url', None)

        if not train_url or not predict_url:
            raise Exception("Error: Both 'train_url' and 'predict_url' must be provided.")

        payload = {
            'df': df.to_json(orient='records'),
            'target': target,
            'args': args
        }
```